### DISCUSSION TITLE
with_polymorphic use string or table name instead of Class

### DISCUSSION THREAD

**Discussion by riwu** [sqlalchemy/sqlalchemy#10871] [Usage Questions]
To avoid circular imports, I cannot import the class to pass to `with_polymorphic`.

Is there a way to pass the table name or the string to `with_polymorphic` instead?  
Eg. `with_polymorphic("User", "*")`

This is possible for relationships: `relationship("User")`

**Comment by CaselIT** [comment]
Hi

with_polymorphic is used to query the data, so it's strange that you have a circular reference at that point, since all the models should be imported at that point

Supporting the proposed feature would be problematic with the current interface, since no registry on classes is passed to it so how can `with_polymorphic("User", "*")` resolve User to the user class.

**Reply by riwu** [reply]
I defined a class method on the model as a utility/helper method, hence the circular import.

Is it not a recommended practice to define class methods in the model classes?

I suppose it could work if you add a way to define polymorphic relationship as a model attribute (like relationship), then the class methods can now reference these attributes within the class?

**Reply by CaselIT** [reply]
If you have a class method can you import the needed model inside the method to avoid a circular reference?

**Reply by riwu** [reply]
Hmm not pretty to have inline imports but I suppose that's a good workaround, thanks!
